nemo/core/classes/modelPT.py

```diff
@@ -739,23 +739,14 @@ def train_dataloader(self):
         if self._train_dl is not None:
             return self._train_dl
 
-    def training_step(self, batch, batch_ix):
```

Comment:
Why are the default methods being removed ? It's important to have them for ease of use while building up models partially (without having to define all of the steps at once)

Reply:
Let me ask you this: Why have them? If we require models to have them, they should be abstract.

As to why we shouldn't have them:
- I removed them and nothing fails, which imo makes these functions useless.
- They aren't up-to-date with LightningModule's latest signature.

I suppose this portion isn't strictly necessary for MelGAN, so I can remove it. But I do feel like this is the correct choice.

Reply:
When developing new models, it's not convenient to have to write every single step immediately - say val and test because they may change significantly. 

Especially in cases where models have multiple eval states - eval during train, vs inference during test etc. 

As to signatures not being updated, that can be updated. 

They aren't abstract as we don't want subclasses to have stubs (as they are now). 

Nothing fails right now because all of the models we developed have already reached end stage of development, it's an issue during development itself. 

I suppose going forward enforcing all subclasses to submit all 3 methods or stubs is fine too.